### DISCUSSION TITLE
Excluding Season00 TV Episode directories

### DISCUSSION THREAD

**Discussion by kenj1** [Taxel/PlexTraktSync#962] [Q&A]
Some TV Shows have special episodes that (on Plex) are included in a Season00 directory but usually never found in tvdb.  For example, I have a "Current TV" library.

Is there a way to exclude *all* Season00 directories?  

Something similar to the following syntax in the config.py??

excluded-libraries:
  - Current TV/*/Season00

**Comment by simonc56** [comment]
Can you give the names of such TV shows ?
The best move is to add the missing episodes on tmdb.

> Is there a way to exclude all Season00 directories?

I don't think so.

**Comment by kenj1** [comment]
There are many TV series that have special video clips (outtakes, interviews, etc) that are typically stored in Season00 on Plex.  Most are not allowed/supported on tvdb, so they're not going to be found during the search.  Hence, a way to tell plextraktsync to ignore them would be useful.

For example, from Game Of Thrones (a screen test of The Hound).  For episodes that are not on tvdb, I create my own episode numbers so Plex incorporates them properly.

Game of Thrones -  s00e109 - Screen Test The Hound.mkv

**Reply by simonc56** [reply]
> There are many TV series that have special video clips (outtakes, interviews, etc) that are typically stored in Season00 on Plex.

You should not store those clips in season 0 folder.
Please read the [Plex Support Articles about Local files and extras](https://support.plex.tv/articles/local-files-for-tv-show-trailers-and-extras/) and organize your library accordingly.

> For episodes that are not on tvdb, I create my own episode numbers so Plex incorporates them properly.

Don't do your own episode numbering with videos that are not episodes. The proper way Plex incorporates those videos is as extras or bonus.

**Comment by simonc56** [comment]
> Most are not allowed/supported on tvdb, so they're not going to be found during the search. 

PlexTraktSync does not search episodes if they have **no external id**. It just skips them.
> Hence, a way to tell plextraktsync to ignore them would be useful.

That's what it does :
https://github.com/Taxel/PlexTraktSync/blob/6aa948ad20e5ba979e46edef76e483e7f3c57c84/plextraktsync/media.py#L183-L186

**Comment by kenj1** [comment]
I was just trying to avoid the spam in the logs because special episodes will never be found.  It would clean up a lot of extra log data... I understand that it may not be a high priority.

**Comment by kenj1** [comment]
I've used Plex since they started and been a Plex Pass user in the beginning.  I've organized this way and this is the way tvdb does it, and what Plex recommends.

https://support.plex.tv/articles/naming-and-organizing-your-tv-show-files/

Your proposed solution, while fitting some of the Plex rules, does not change the end result with PlexTraktSync since those episodes still will not be found in tvdb and will cause unnecessary spam in the log file.  I understand it's not high priority since no functionality is impacted.  I will live with the spam.  It was just a simple question of whether it was supported or not.  The answer is no.

**Reply by simonc56** [reply]
> Your proposed solution, while fitting some of the Plex rules, does not change the end result

Yes it does change the end result. Respecting Plex [file organisation](https://support.plex.tv/articles/local-files-for-tv-show-trailers-and-extras/) is needed to have clean logs because extras and bonus are not processed by PlexTraktSync.

> those episodes still will not be found in tvdb

Those are not episodes, they are extras and bonus.

> and will cause unnecessary spam in the log file

No, because not processed during a sync. Only episodes and movies are processed.

> this is the way tvdb does it, and what Plex recommends.

Plex talks about **episodes** (**not extras and bonus**) to place in Season 00 :

`If an “episode” you have doesn’t appear in TheTVDB (e.g. DVD Specials or goof reel), place this content in the Season 00 or Specials`

To finish, the recommandation is to use [TMDB](https://blog.trakt.tv/tv-show-metadata-e6e64ed4e6ef) as primary source for episodes numbering.

**Comment by kenj1** [comment]
I'm seeing that TMDB is a better source than TVDB for some but not for all.  Wrt Plex, DVD specials typically include "behind the scenes", "deleted scenes", etc.  I think Plex offers two methods for handling of these special files.  I'll have to live with the spam for now and transition to TMDB (or TVDB depending on TMDB's accuracy) for the TV Shows.
